tests/nfs_create_retain_pv_disable_exporter_test.go

```diff
@@ -0,0 +1,329 @@
+
+	When("Cleaning up NFS PV", func() {
+		It("should delete nfs pv", func() {
+			Expect(nfsPVName).NotTo(BeEmpty(), "nfsPVName should not be empty")
+
+			err := Client.deletePV(nfsPVName)
+			Expect(err).To(BeNil(), "while deleting NFS pv %s", nfsPVName)
+
+			var isNFSPVDeleted bool
+			for retry := 0; retry < maxRetryCount; retry++ {
+				_, err = Client.getPV(nfsPVName)
+				if err != nil && k8serrors.IsNotFound(err) {
+					isNFSPVDeleted = true
+					break
+				}
+				Expect(err).To(BeNil(), "while fetching NFS pv %s", nfsPVName)
+				time.Sleep(time.Second * 5)
+			}
+			Expect(isNFSPVDeleted).To(BeTrue(), "NFS pv %s should be deleted", nfsPVName)
+		})
+
+		It("should send events to server", func() {
+			Expect(backendPVCName).NotTo(BeEmpty(), "backendPVCName should not be empty")
+			Expect(nfsPVName).NotTo(BeEmpty(), "nfsPVName should not be empty")
+			Expect(backendPVName).NotTo(BeEmpty(), "backendPVName should not be empty")
+
+			var backendPVCObj *corev1.PersistentVolumeClaim
+			var err error
+			var isEventReceived bool
+
+			for retry := 0; retry < maxRetryCount; retry++ {
+				backendPVCObj, err = Client.getPVC(OpenEBSNamespace, backendPVCName)
+				Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+				_, isEventReceived = backendPVCObj.Annotations[nfs.VolumeDeleteNFSPVKey]
+				if isEventReceived {
+					break
+				}
+				time.Sleep(time.Second * 10)
+			}
+			Expect(isEventReceived).To(BeTrue(), "NFS pv %s details are not exported to server for delete event", nfsPVName)
+			Expect(backendPVCObj.Annotations[nfs.VolumeDeleteNFSPVKey]).To(Equal(nfsPVName), "while verifying NFS pv delete event data")
+			Expect(backendPVCObj.Annotations[nfs.VolumeDeleteBackendPVCKey]).To(Equal(OpenEBSNamespace+"-"+backendPVCName), "while verifying backend pvc delete event data")
+			Expect(backendPVCObj.Annotations[nfs.VolumeDeleteBackendPVKey]).To(Equal(backendPVName), "while verifying backend pv delete event data")
+		})
+	})
+
+	When("Test event finalizers are removed from nfs resources", func() {
+		It("should delete nfs resources", func() {
+			Expect(backendPVCName).NotTo(BeEmpty(), "backendPVCName should not be empty")
+
+			// Remove test finalizer on Backend PVC
+			backendPVC, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
+			Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+			removeFinalizer(&backendPVC.ObjectMeta, integrationTestFinalizer)
+
+			_, err = Client.updatePVC(backendPVC)
+			Expect(err).To(BeNil(), "while removing test protection finalizer on backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+			err = Client.deleteDeployment(OpenEBSNamespace, backendPVCName)
+			Expect(err).To(BeNil(), "while deleting deployment %s/%s", OpenEBSNamespace, backendPVCName)
+
+			err = Client.deletePVC(OpenEBSNamespace, backendPVCName)
+			Expect(err).To(BeNil(), "while deleting backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
```

Comment:
We also need to delete the NFS Service resource (or) let's leave it to garbage collector

Reply:
we can add that since it is only missing. we can leave it to garbage collector also but may conflict if the future test is having the same pvc name. I'll add deletion for service